### PR TITLE
Add `--trigger-width` support for DatePicker & DateRangePicker

### PR DESCRIPTION
**Pull request by sookmax**
Closes https://github.com/adobe/react-spectrum/issues/5883

Hello, this PR adds `--trigger-width` CSS variable as a popover inline style, so that users can match the width of the popover containing the calendar to that of the input group with said variable. 

The changes contained in this PR make direct references to those components already well supporting the `--trigger-width` such as [Select](https://github.com/adobe/react-spectrum/blob/main/packages/react-aria-components/src/Select.tsx#L96-L107), [Menu](https://github.com/adobe/react-spectrum/blob/main/packages/react-aria-components/src/Menu.tsx#L48-L58), and [Combobox](https://github.com/adobe/react-spectrum/blob/main/packages/react-aria-components/src/ComboBox.tsx#L149-L164).

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

#### DatePicker
http://localhost:9003/?path=/story/react-aria-components--date-picker-trigger-width-example&providerSwitcher-express=false&strict=true
#### DateRangePicker
http://localhost:9003/?path=/story/react-aria-components--date-range-picker-trigger-width-example&providerSwitcher-express=false&strict=true

### CODE REVIEW

**Comment by reidbarber** [comment]
GET_BUILD

**Review by reidbarber — APPROVED** [review]
LGTM, thanks!

File: packages/react-aria-components/stories/DatePicker.stories.tsx [hunk 1]
```diff
@@ -52,6 +52,40 @@ export const DatePickerExample = () => (
...
+      <Button>🗓</Button>
+    </Group>
+    <Popover
+      placement="bottom start"
+      style={{
+        background: 'Canvas',
+        color: 'CanvasText',
+        border: '1px solid gray',
+        padding: 20,
+        width: 'calc(var(--trigger-width) - 20px * 2 - 1px * 2)'
```
Comment (LFDanLu) [inline packages/react-aria-components/stories/DatePicker.stories.tsx:71]: A bit unfortunate that the other popover styles force a calc to be a needed haha

**Review by LFDanLu — APPROVED** [review]
Thanks for the speedy fix!

**Review by snowystinger — APPROVED** [review]
Thanks for the PR! I hope you don't mind, I simplified the width calculation.

**Comment by sookmax** [comment]
Haha border-box is brilliant. Thanks!

**Comment by snowystinger** [comment]
@Emiliano-Bucci you can set this up for any popover in the same way. It's not built into popover because the trigger may be any number of things which are specific to the implementing component and a Popover does not contain any information about its trigger.

This has been added in relevant components such as Combobox, Menu, and Select. And thanks to this PR it will now be available for DatePicker and DateRangePicker.

**Pull request merged**